### PR TITLE
[#ENTE-68] Create io-functions-services pipelines

### PR DESCRIPTION
**Pull request by pasqualedevita**
this pr create pipelines for new azure function https://github.com/pagopa/io-functions-services-cache

### CODE REVIEW

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  }
+}
+
+# code review
+resource "azuredevops_build_definition" "io-functions-services-cache-code-review" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-pr, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-services-cache.repository.name}.code-review"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:22]:
```suggestion
  name       = format("%s.code-review", var.io-functions-services-cache.repository.name)
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  path       = "\\${var.io-functions-services-cache.repository.name}"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:23]:
```suggestion
  path       = format("\\%s", var.io-functions-services-cache.repository.name)
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      path_filter {
+        exclude = []
+        include = []
+      }
+    }
+  }
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-functions-services-cache.repository.organization}/${var.io-functions-services-cache.repository.name}"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:45]:
```suggestion
    repo_id               = join("/", [var.io-functions-services-cache.repository.organization, var.io-functions-services-cache.repository.name] )
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    branch_name           = var.io-functions-services-cache.repository.branch_name
+    yml_path              = "${var.io-functions-services-cache.repository.pipelines_path}/code-review-pipelines.yml"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:47]:
```suggestion
    yml_path              = join("/", [var.io-functions-services-cache.repository.pipelines_path, "code-review-pipelines.yml"])
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  authorized    = true
+  type          = "endpoint"
+}
+
+# deploy
+resource "azuredevops_build_definition" "io-functions-services-cache-deploy" {
+  depends_on = [azuredevops_serviceendpoint_github.io-azure-devops-github-rw, azuredevops_serviceendpoint_azurerm.PROD-IO, azuredevops_project.project]
+
+  project_id = azuredevops_project.project.id
+  name       = "${var.io-functions-services-cache.repository.name}.deploy"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:73]:
```suggestion
  name       = format("%s.deploy", var.io-functions-services-cache.repository.name)
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  path       = "\\${var.io-functions-services-cache.repository.name}"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:74]:
```suggestion
  path       = format("\\%s", var.io-functions-services-cache.repository.name)
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+  repository {
+    repo_type             = "GitHub"
+    repo_id               = "${var.io-functions-services-cache.repository.organization}/${var.io-functions-services-cache.repository.name}"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:78]:
```suggestion
    repo_id               = join("/", [var.io-functions-services-cache.repository.organization, var.io-functions-services-cache.repository.name ])
```

File: azure-devops/projects/io-backend-projects/io-functions-services-cache.tf [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    branch_name           = var.io-functions-services-cache.repository.branch_name
+    yml_path              = "${var.io-functions-services-cache.repository.pipelines_path}/deploy-pipelines.yml"
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/io-functions-services-cache.tf:80]:
```suggestion
    yml_path              = join("/", [var.io-functions-services-cache.repository.pipelines_path, "deploy-pipelines.yml"])
```

**Review by uolter — CHANGES_REQUESTED** [review]

**Comment by uolter** [comment]
@pasqualedevita by the way it's up to you .... it's not wrong.... just a style exercise. :-)

**Review by uolter — APPROVED** [review]

**Comment by pasqualedevita** [comment]
> @pasqualedevita by the way it's up to you .... it's not wrong.... just a style exercise. :-)

if is only code style I prefer inline

**Pull request merged**